--- Makefile.in	2021-09-21 15:26:03.801953500 -0400
+++ Makefile.in	2021-09-21 15:12:56.992609200 -0400
@@ -351,7 +351,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 # Order is important:
-SUBDIRS = third_party src testdata examples msvc xcode
+SUBDIRS = third_party src testdata msvc xcode
 EXTRA_DIST = \
 	AUTHORS \
 	COPYING \
--- third_party/Makefile.in	2021-09-21 15:26:22.358766400 -0400
+++ third_party/Makefile.in	2021-09-21 15:21:27.220391400 -0400
@@ -443,7 +443,7 @@
 	      -I$(top_srcdir)/third_party/googletest-r108/include
 
 lib_LTLIBRARIES = libminizip.la liburiparser.la
-noinst_LTLIBRARIES = libgtest.la libgtest_main.la
+#noinst_LTLIBRARIES = libgtest.la libgtest_main.la
 minizip_root = zlib-1.2.3
 minizip = $(minizip_root)/contrib/minizip
 minizipinc = $(includedir)/kml/third_party/$(minizip)
@@ -717,10 +717,10 @@
 	  rm -f $${locs}; \
 	}
 $(googletest)/src/$(am__dirstamp):
-	@$(MKDIR_P) $(googletest)/src
+	@$(MKDIR_P) googletest-r108/src
 	@: > $(googletest)/src/$(am__dirstamp)
 $(googletest)/src/$(DEPDIR)/$(am__dirstamp):
-	@$(MKDIR_P) $(googletest)/src/$(DEPDIR)
+	@$(MKDIR_P) googletest-r108/src/$(DEPDIR)
 	@: > $(googletest)/src/$(DEPDIR)/$(am__dirstamp)
 $(googletest)/src/gtest.lo: $(googletest)/src/$(am__dirstamp) \
 	$(googletest)/src/$(DEPDIR)/$(am__dirstamp)
@@ -764,10 +764,10 @@
 libminizip.la: $(libminizip_la_OBJECTS) $(libminizip_la_DEPENDENCIES) $(EXTRA_libminizip_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libminizip_la_LINK) -rpath $(libdir) $(libminizip_la_OBJECTS) $(libminizip_la_LIBADD) $(LIBS)
 $(uriparser)/lib/$(am__dirstamp):
-	@$(MKDIR_P) $(uriparser)/lib
+	@$(MKDIR_P) uriparser-0.7.5/lib
 	@: > $(uriparser)/lib/$(am__dirstamp)
 $(uriparser)/lib/$(DEPDIR)/$(am__dirstamp):
-	@$(MKDIR_P) $(uriparser)/lib/$(DEPDIR)
+	@$(MKDIR_P) uriparser-0.7.5/lib/$(DEPDIR)
 	@: > $(uriparser)/lib/$(DEPDIR)/$(am__dirstamp)
 $(uriparser)/lib/UriCommon.lo: $(uriparser)/lib/$(am__dirstamp) \
 	$(uriparser)/lib/$(DEPDIR)/$(am__dirstamp)
