diff -ubr libmicrohttpd-0.9.46/w32/VS2013/libmicrohttpd.vcxproj libmicrohttpd-0.9.46/w32.pgsc/VS2013/libmicrohttpd.vcxproj
--- libmicrohttpd-0.9.46/w32/VS2013/libmicrohttpd.vcxproj	2015-10-02 08:33:51.000000000 -0400
+++ libmicrohttpd-0.9.46-pgsc/w32/VS2013/libmicrohttpd.vcxproj	2016-02-11 16:09:52.788683400 -0500
@@ -331,8 +331,8 @@
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(ConfigurationType)'=='StaticLibrary'">
     <ClCompile>
-      <RuntimeLibrary Condition="'$(UseDebugLibraries)'=='true'">MultiThreadedDebug</RuntimeLibrary>
-      <RuntimeLibrary Condition="'$(UseDebugLibraries)'!='true'">MultiThreaded</RuntimeLibrary>
+      <RuntimeLibrary Condition="'$(UseDebugLibraries)'=='true'">MultiThreadedDebugDLL</RuntimeLibrary>
+      <RuntimeLibrary Condition="'$(UseDebugLibraries)'!='true'">MultiThreadedDLL</RuntimeLibrary>
     </ClCompile>
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(ConfigurationType)'=='DynamicLibrary'">
